### PR TITLE
Symfonize agenda

### PR DESCRIPTION
**Pull request by sillydebs**

### CODE REVIEW

File: public/planningHebdo/class.planningHebdo.php [hunk 1]
```diff
@@ -16,9 +16,10 @@
 */
 
 // pas de $version=acces direct aux pages de ce dossier => Accès refusé
-if (!isset($version) and php_sapi_name() != 'cli') {
-    include_once __DIR__."/../include/accessDenied.php";
-}
+
+//if (!isset($version) and php_sapi_name() != 'cli') {
+//    include_once __DIR__."/../include/accessDenied.php";
+//}
```
Comment (alexarn) [inline public/planningHebdo/class.planningHebdo.php:22]: Code to remove ?

File: src/Controller/CalendarController.php [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+require_once(__DIR__.'/../../public/joursFeries/class.joursFeries.php');
+require_once(__DIR__.'/../../public/personnel/class.personnel.php');
+
+class CalendarController extends BaseController
+{
+    /**
+     * @Route("calendar", name = "calendar.index", methods={"GET"})
+     */
+
```
Comment (alexarn) [inline src/Controller/CalendarController.php:19]: Blank line between annotation and function

File: public/setup/atomicupdate/update_path_calendar.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+
+$sql[]="UPDATE `{$dbprefix}acces` SET `page`='/calendar' WHERE `page`='agenda/index.php';";
+$sql[]="UPDATE `{$dbprefix}menu` SET `url`='/calendar' WHERE `url`='agenda/index.php';";
```
Comment (alexarn) [inline public/setup/atomicupdate/update_path_calendar.php:5]: That is ok, but these changes are missing in setup/db_data.php

File: src/Controller/CalendarController.php [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    public function index(Request $request, Session $session){
+        $debut = $request->get('debut');
+        $fin = $request->get('fin');
+
+        $debut = filter_var($debut, FILTER_CALLBACK, array("options"=>"sanitize_dateFr"));
+        $fin = filter_var($fin, FILTER_CALLBACK, array("options"=>"sanitize_dateFr"));
```
Comment (alexarn) [inline src/Controller/CalendarController.php:25]: This is useless

File: src/Controller/CalendarController.php [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+        if (!array_key_exists('agenda_debut', $_SESSION)){
+            $_SESSION['agenda_debut']=null;
+            $_SESSION['agenda_fin']=null;
+            $_SESSION['agenda_perso_id']=$_SESSION['login_id'];
+        }
```
Comment (alexarn) [inline src/Controller/CalendarController.php:31]: That don't make an exepected behavior. Start/End date are not saved in session

File: src/Controller/CalendarController.php [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+        $debut=$debut?$debut:$_SESSION['agenda-debut'];
+        $fin=$fin?$fin:$_SESSION['agenda-fin'];
```
Comment (alexarn) [inline src/Controller/CalendarController.php:34]: Please add spaces in your code

File: src/Controller/CalendarController.php [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+
+        $admin = in_array(3, $GLOBALS['droits'])?true:false;
```
Comment (alexarn) [inline src/Controller/CalendarController.php:36]: Spaces please

File: src/Controller/CalendarController.php [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+        $fin = dateFr3($finSQL);
+        $_SESSION['agenda_debut']=$debut;
+        $_SESSION['agenda_fin']=$fin;
+        $_SESSION['agenda_perso_id']=$perso_id;
+        $class = null;
+
+        $nonValides = $this->config('Agenda-Plannings-Non-Valides');
+
+        //PlanningHebdo et EDTSamedi étant incompatibles, EDTSamedi est désactivé si PlanningHebdo est activé
+
```
Comment (alexarn) [inline src/Controller/CalendarController.php:57]: Useless blank line

File: src/Controller/CalendarController.php [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+        if($this->config('PlanningHebdo')){
+            $this->config('EDTSamedi',0);
+        }
+
+        //Sélection du personnel pour le menu déroulant
+        $toutlemonde= $this->config('toutlemonde')?null:"AND id<>2 ";
```
Comment (alexarn) [inline src/Controller/CalendarController.php:63]: Remove this please. Its a old bug

File: src/Controller/CalendarController.php [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+                if ($d->semaine3==2) {
+                    $jour=$jour+7;
+                } elseif ($d->semaine3==3) {
+                    $jour=$jour+14;
+                }
+            }
+
+            // Si utilisation d'un planning pour les semaines sans samedi et un planning pour les semaines avec samedi travaillé
+            if ($this->config('EDTSamedi')) {
+              // Pour chaque agent, recherche si la semaine courante est avec samedi travaillé ou non
```
Comment (alexarn) [inline src/Controller/CalendarController.php:157]: Indentation

File: src/Controller/CalendarController.php [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+                $p=new \personnel();
+                $p->fetchEDTSamedi($perso_id, $j1, $j1);
+                $jour += $p->offset;
+            }
+
+             //ø Horaires de travail si le module PlanningHebdo est activé
+            if ($this->config('PlanningHebdo')) {
+            include_once __DIR__."/../../public/planningHebdo/class.planningHebdo.php";
```
Comment (alexarn) [inline src/Controller/CalendarController.php:165]: indentation

**Review by alexarn — CHANGES_REQUESTED** [review]

**Comment by alexarn** [comment]
File public/agenda/index.php should be removed

File: public/include/feries.php [hunk 12]
```diff
@@ -18,11 +18,12 @@
...
-    include_once "accessDenied.php";
-    exit;
-}
-
+$version = $GLOBALS['version'] ?? null;
+//if (!isset($version)) {
+//    include_once "accessDenied.php";
+//    exit;
+//}
+//
```
[the remark this answers is not in the document]
Comment (sillydebs) [reply public/include/feries.php:26]: I have reviewed this with @jeromecombes. This file was either required or included in other files (it had to) but this part of code was keeping me from doing it.

File: src/Controller/CalendarController.php (hunk 10 above)
Comment (sillydebs) [reply src/Controller/CalendarController.php:157]: Which element is concerned in this extract ?

**Review by sillydebs — COMMENTED** [review]
This commit doesn't include comments that I didn't understand accurately.

File: public/include/feries.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (alexarn) [reply public/include/feries.php:26]: Either you uncomment this part (if needed) or you don't add it. Please don't add commented code.

File: src/Controller/CalendarController.php [hunk 13]
```diff
@@ -0,0 +1,407 @@
...
+        $class = null;
+        $nonValides = $this->config('Agenda-Plannings-Non-Valides');
+
+        //PlanningHebdo et EDTSamedi étant incompatibles, EDTSamedi est désactivé si PlanningHebdo est activé
+        if($this->config('PlanningHebdo')){
+            $this->config('EDTSamedi',0);
+        }
+
+        //Sélection du personnel pour le menu déroulant
+
```
Comment (alexarn) [inline src/Controller/CalendarController.php:57]: blank line after comment

**Review by alexarn — CHANGES_REQUESTED** [review]

**Comment by alexarn** [comment]
Everybody still here in the agents list. Please remove it

**Comment by alexarn** [comment]
Merged on master

**Pull request closed without merging**